src/utils/AssayGeneratorMetabo.cpp

```diff
@@ -0,0 +1,523 @@
+
+  If the FFM featureXML is used the "use_known_unknowns" flag is used automatically.
+
+  Internal procedure AssayGeneratorMetabo:
+  1. Input mzML and featureXML
+  2. Annotate precursor mz and intensity
+  3. Filter feature by convexhull size
+  4. Assign precursors to specific feature
+  5. Extract feature meta information (if possible)
+  6. Find MS2 spectrum with highest intensity precursor for one feature
+  7. Dependent on the method use the MS2 with the highest intensity precursor or a consensus spectrum
+     for the transition calculation
+  8. Calculate thresholds (maximum and minimum intensity for transition peak)
+  9. Extract and write transitions
+
+  <B>The command line parameters of this tool are:</B>
+  @verbinclude UTILS_SiriusAdapter.cli
+  <B>INI file documentation of this tool:</B>
+  @htmlinclude UTILS_SiriusAdapter.html
+ */
+
+/// @cond TOPPCLASSES
+
+
+/// struct to hold assay information of one row 
+struct AssayRow
+{
+  double precursor_mz;
+  double product_mz;
+  float library_int;
+  double normalized_rt;
+  String compound_name;
+  String smiles;
+  String sumformula;
+  String adduct;
+  String transition_group_id;
+  String transition_id;
+  bool decoy; 
+};
+
+class TOPPAssayGeneratorMetabo :
+  public TOPPBase
+{
+public:
+  TOPPAssayGeneratorMetabo() :
+    TOPPBase("AssayGeneratorMetabo", "Assay library generation from DDA data (Metabolomics)", false)
+    {}
+
+protected:
+
+  void registerOptionsAndFlags_() override
+  {
+
+    registerInputFile_("in", "<file>", "", "MzML Input file");
+    setValidFormats_("in", ListUtils::create<String>("mzml"));
+
+    registerInputFile_("in_id", "<file>", "", "FeatureXML Input with id information (accurate mass search)");
+    setValidFormats_("in_id", ListUtils::create<String>("featurexml"));
+
+    registerOutputFile_("out", "<file>", "", "Assay library output file");
+    setValidFormats_("out", ListUtils::create<String>("tsv"));
+
+    registerStringOption_("method", "<choice>", "highest_intensity", "",false);
+    setValidStrings_("method", ListUtils::create<String>("highest_intensity,consensus_spectrum"));
+
+    registerDoubleOption_("precursor_mz_tolerance", "<num>", 0.005, "Tolerance window (left and right) for precursor selection", false);
+    registerDoubleOption_("precursor_rt_tolerance", "<num>", 5, "Tolerance window (left and right) for precursor selection", false);
+
+    registerDoubleOption_("cosine_similarity_threshold", "<num>", 0.98, "Threshold for cosine similarity of MS2 spectras of same precursor used for consensus spectrum", false);
+
+    registerIntOption_("filter_by_convex_hulls", "<num>", 2, "Features have to have at least x MassTraces", false);
+
+    registerDoubleOption_("transition_threshold", "<num>", 10, "Further transitions need at least x% of the maximum intensity (default 10%)", false);
+
+    registerFlag_("use_known_unknowns", "Use features without identification", false);
```

Comment:
do you mean spectra without identification? how else are you getting transitions?

Reply:
I mean that a feature was found in the FeatureFinderMetabo but no identification was performed on that feature beforehand (like AccurateMassSearch or de-novo identification). Which would mean that It is a compound (feature) but you do not know which one - known_unknown).